tools/unused.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import b
+
+# @run_on_docker
+def register(command):
+    # Install help
+    command.help = "Creates a list of unused or commented out modules"
+
+
+# @run_on_docker
+def run(**kwargs):
+
+    source_path = b.cmake_cache["NUbots_SOURCE_DIR"]
+    roles_path = source_path + "/roles"
```

Comment:
When joining paths in python you should use `os.path.join('a', 'b', 'c', 'd')`
```python
    roles_path = os.path.join(source_path, "/roles")
```
That said the roles path should be another variable in the cmake_cache. However, it must have been missed and should be added to `nuclear/CMakeLists.txt` and used in `nuclear/roles/CMakeLists.txt`
and then you can use
```suggestion
    roles_path = os.path.join(source_path + b.cmake_cache['NUCLEAR_ROLE_DIR'])
```